### PR TITLE
Homework2: Yauheni Chachukha

### PR DESCRIPTION
**Pull request by chechuhae**

### CODE REVIEW

File: src/Fibonacchi.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"""
+Выведите первые n чисел Фибоначчи, (используйте временные переменные для
+хранения прошлых значений и цикл for)
+"""
+
+qun = int(input("Введите число более 0: \n"))  # количество элементов ряда
+number1 = int(0)  # 1 элемент ряда
```
Comment (belovmd) [inline src/Fibonacchi.py]: = 0
= 1

File: src/Fibonacchi.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+if qun == 1:  # один элемент в ряде
+    print(number1)
+elif qun == 2:  # два элемент в ряде
+    print(number1, number2)
+else:  # более двух
+    print(number1, end=' ')
+    print(number2, end=' ')
+    for i in range(2, qun):
+        time = number1 + number2
+        number1 = number2
```
Comment (belovmd) [inline src/Fibonacchi.py]: number1, number2 = number2, number2+number1

File: src/Sum.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    newrub = rub
+    newkop = kop
+    price = float(newrub + newkop / 100)
+# quntaty of purchase
+qun = int(input("Введите количество, чтобы расчитать сумму покупки: "))
+sumrub = int(price * qun // 1)  # summ in rubles
+endrub = (sumrub - (sumrub // 1000) * 1000)
+# identify rubles word endding
+end1 = ""
+if endrub % 10 == 0:
```
Comment (belovmd) [inline src/Sum.py:32]: great

File: src/Sum.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+цена, а также количество L товара Посчитайте общую цену в рублях и копейках за
+L товаров.
+Пример:
+Input: Цена одной вещи 3 рубля 20 копеек, посчитать 3 предмета.
+Output: Общая цена 9 рублей 60 копеек
+"""
+
+rub = int(input("Введите количество рублей: "))  # rubles
+kop = int(input("Введите количество копеек: "))  # pennies
+if len(str(kop)) >= 3:  # if len pennies > 2 symbols
```
Comment (belovmd) [inline src/Sum.py:18]: It's much more easy to work only with kopecks (rouble ->kopecks) and than find them.

File: src/Sum.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    newrub = rub + kop // (10 ** (len(str(kop)) - 1))
+    newkop = kop % 10 ** (len(str(kop)) - 1)
+    price = float(newrub + newkop / 100)
```
Comment (belovmd) [inline src/Sum.py]: we don't need float here. another division would be enough

File: src/Simple.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+"""
+Проверить, является ли число простым
+"""
+
+n = int(input("Введите число: \n"))  # enter number
+c = n  # number 2
```
Comment (belovmd) [inline src/Simple.py]: c n - bad names

File: src/Simple.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+while n > 1:
+    n -= 1  # division number 2 on number 1 -= 1
+    if c % n == 0 and n != 1:
```
Comment (belovmd) [inline src/Simple.py]: we can check only multipliers from 2 to sqrt(n)

File: src/Palindrom.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+"""
+
+number = int(input())  # присваиваем значение числа
+number2 = number  # присваиваем значение числа 1 числу 2
+lenstr = 0  # присваиваем значение 0 длине числа
+while number != 0:  # считаем длину числа
+    lenstr += 1
+    number //= 10
+lenstr2 = lenstr  # длина числа для цикла
+while lenstr2 > 1 or number != 0:  # сравниваем попарно, обрезаем
```
Comment (belovmd) [inline src/Palindrom.py]: there is much more simple way using division by 10

File: src/Longinsentence.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
+"""
+Найти самое длинное слово в введенном предложении. Учтите что в предложении
+есть знаки препинания.
+Подсказки:
+my_string.split([chars]) возвращает список строк.
+len(list) - количество элементов в списке
+"""
+main = str(input("Прошу ввести предложение для разложения:"))
+pun = ",.-'"'"'"?!:;""''"  # punctuation marks
```
Comment (belovmd) [inline src/Longinsentence.py]: too many ''''''' isn't it?
what about {}[];: ?

File: src/Longinsentence.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+k = 0  # number of mark in string
+while k != len(pun):  # replace all punctuation in maim string
```
Comment (belovmd) [inline src/Longinsentence.py:24]: for k in pun: string.replace(k,"")?

File: src/Longinsentence.py [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+timelen = len(timemain)  # temporary len
+k = 0  # number 2 of mark in string
+super = timemain[k]  # assignment super 0 in str time main
+while k != timelen - 1:
+    if len(super) >= len((timemain[k + 1])):
+        super = super
+    else:
+        super = timemain[k + 1]
+    k += 1
+print(super)
```
Comment (belovmd) [inline src/Longinsentence.py]: "super"
super if it's not a string is reserved word

File: src/HR1.py [hunk 12]
```diff
@@ -0,0 +1,10 @@
+def solveMeFirst(a, b):
```
Comment (belovmd) [inline src/HR1.py]: please put task description

File: src/HR2.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
+"""
+Task
+Given an integer,n, perform the following conditional actions:
+If n is odd, print Weird
+If nis even and in the inclusive range of 2 to 5, print Not Weird
```
Comment (belovmd) [inline src/HR2.py]: n is - mispring

File: src/HR2.py [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+n = int(input("Введите число:\n"))
+if n % 2 == 0 and 6 <= n <= 20:
+    print("Weird")
+elif n % 2 == 0 and 2 <= n <= 5:
+    print("Not Weird")
+elif n % 2 == 0 and n > 20:
+    print("Not Weird")
+elif n % 2 != 0:
+    print("Weird")
```
Comment (belovmd) [inline src/HR2.py:19]: `if n%2:1`
`if <n<:`
`elif`
(nested if)

**Review by belovmd — COMMENTED** [review]
1. It seems all these tasks were elementary level. you can solve some more difficult if you want
2. please address next issues

File: src/HR2.py (hunk 14 above)
Comment (chechuhae) [reply src/HR2.py:19]: changed files in accordance with all comments.

File: src/Longinsentence.py (hunk 10 above)
Comment (belovmd) [reply src/Longinsentence.py:24]: why not suggested option?

File: src/Simple.py [hunk 15]
```diff
@@ -0,0 +1,12 @@
+"""
+Проверить, является ли число простым
+"""
+
+number = int(input("Введите число: \n"))  # enter number
+end = number // 2 + 1
+for i in range(2, end):
```
Comment (belovmd) [inline src/Simple.py:7]: sqrt(number)

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**